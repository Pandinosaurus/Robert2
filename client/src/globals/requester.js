import axios from 'axios';
import deepObjectSet from 'lodash/set';
import isPlainObject from 'lodash/isPlainObject';
import flattenObject from '@/utils/flattenObject';
import config from '@/globals/config';
import cookies from '@/utils/cookies';
import DateTime from '@/utils/datetime';
import Day from '@/utils/day';

const requester = axios.create({
    baseURL: config.api.url,
    headers: config.api.headers,
});

requester.interceptors.request.use(
    (_request) => {
        const { params, onProgress, ...request } = _request;

        // - Ajoute une méthode haut-niveau permettant de récupérer l'avancement sous forme numérique.
        //   - Si la méthode de la requête est POST, PUT ou PATCH, c'est l'événement `onUploadProgress` qui sera écouté.
        //   - Sinon ce sera `onDownloadProgress`
        if (onProgress) {
            const progressCallback = (event) => {
                if (!event.lengthComputable) {
                    return;
                }

                const { loaded, total } = event;
                const percent = (loaded / total) * 100;
                onProgress(percent);
            };

            if (request.method && ['POST', 'PUT', 'PATCH'].includes(request.method.toUpperCase())) {
                request.onUploadProgress = progressCallback;
            } else {
                request.onDownloadProgress = progressCallback;
            }
        }

        // - Traitement spécial des objets littéraux contenant des fichiers.
        //   => On utilise un envoi via `multipart/form-data` et on met les données
        //      supplémentaires éventuelles dans une clé spéciale `@data`)
        const hasFiles = (
            isPlainObject(request.data) &&
            Object.values(request.data).some((value) => (
                value instanceof File
            ))
        );
        if (hasFiles) {
            const formData = new FormData();
            const data = Object.entries(flattenObject(request.data)).reduce(
                (_data, [key, value]) => {
                    if (value instanceof File) {
                        formData.append(btoa(key), value);
                        return _data;
                    }
                    return deepObjectSet(_data, key, value);
                },
                {},
            );

            formData.append('@data', JSON.stringify(data));
            request.data = formData;
        }

        if (params) {
            Object.keys(params).forEach((name) => {
                if (params[name] === true) {
                    params[name] = '1';
                }
                if (params[name] === false) {
                    params[name] = '0';
                }
                if (params[name] instanceof DateTime || params[name] instanceof Day) {
                    params[name] = params[name].toString();
                }
            });
            request.params = params;
        }

        const authToken = cookies.get(config.auth.cookie);
        if (authToken) {
            request.headers.Authorization = `Bearer ${authToken}`;
        }
        return request;
    },
    (error) => Promise.reject(error),
);

export default requester;
