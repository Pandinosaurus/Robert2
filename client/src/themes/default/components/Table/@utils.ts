import type { ColumnsVisibility } from 'vue-tables-2-premium';

export type ColumnsDisplay = Record<string, boolean>;

const STORAGE_KEY_PREFIX = 'vuetables_';

const initFromStorage = (tableName: string, columns: ColumnsDisplay): ColumnsDisplay => {
    const storedTableState = localStorage.getItem(`${STORAGE_KEY_PREFIX}${tableName}`);
    if (!storedTableState) {
        return columns;
    }

    let tableState;
    try {
        tableState = JSON.parse(storedTableState);
    } catch {
        localStorage.removeItem(`${STORAGE_KEY_PREFIX}${tableName}`);
        return columns;
    }

    if (!tableState) {
        return columns;
    }

    const { userColumnsDisplay } = tableState;
    if (!userColumnsDisplay) {
        return columns;
    }

    return Object.fromEntries(
        Object.keys(columns).map((column: string) => (
            [column, userColumnsDisplay.includes(column)]
        )),
    );
};

export const initColumnsDisplay = (tableName: string | undefined, columns: ColumnsDisplay): ColumnsVisibility => (
    Object.fromEntries(
        Object
            .entries((
                tableName !== undefined
                    ? initFromStorage(tableName, columns)
                    : columns
            ))
            .map(([name, visible]: [string, boolean]) => (
                [name, `${visible ? 'min' : 'max'}_mobileP`]
            )),
    )
);
