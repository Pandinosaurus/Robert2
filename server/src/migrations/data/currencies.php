<?php
declare(strict_types=1);

return [
    // ------------------------------------------------------
    // -
    // -    Francophones
    // -
    // ------------------------------------------------------

    ['code' => 'EUR', 'name' => 'Euro'],
    ['code' => 'CHF', 'name' => 'Franc suisse'],
    ['code' => 'CAD', 'name' => 'Dollar canadien'],
    ['code' => 'BIF', 'name' => 'Franc burundais'],
    ['code' => 'CDF', 'name' => 'Franc congolais'],
    ['code' => 'DJF', 'name' => 'Franc djiboutien'],
    ['code' => 'DZD', 'name' => 'Dinar algérien'],
    ['code' => 'GNF', 'name' => 'Franc guinéen'],
    ['code' => 'KMF', 'name' => 'Franc comorien'],
    ['code' => 'MAD', 'name' => 'Dirham marocain'],
    ['code' => 'RWF', 'name' => 'Franc rwandais'],
    ['code' => 'TND', 'name' => 'Dinar tunisien'],
    ['code' => 'XAF', 'name' => 'Franc CFA (BEAC)'],
    ['code' => 'XOF', 'name' => 'Franc CFA (BCEAO)'],
    ['code' => 'XPF', 'name' => 'Franc pacifique'],

    // ------------------------------------------------------
    // -
    // -    Non francophones
    // -
    // ------------------------------------------------------

    ['code' => 'AFN', 'name' => "Afghani afghan"],
    ['code' => 'MGA', 'name' => "Ariary malgache"],
    ['code' => 'THB', 'name' => "Baht thaïlandais"],
    ['code' => 'PAB', 'name' => "Balboa panaméen"],
    ['code' => 'ETB', 'name' => "Birr éthiopien"],
    ['code' => 'VES', 'name' => "Bolívar vénézuélien"],
    ['code' => 'BOB', 'name' => "Boliviano bolivien"],
    ['code' => 'GHS', 'name' => "Cedi ghanéen"],
    ['code' => 'CRC', 'name' => "Colón costaricien"],
    ['code' => 'NIO', 'name' => "Córdoba nicaraguayen"],
    ['code' => 'DKK', 'name' => "Couronne danoise"],
    ['code' => 'NOK', 'name' => "Couronne norvégienne"],
    ['code' => 'SEK', 'name' => "Couronne suédoise"],
    ['code' => 'CZK', 'name' => "Couronne tchèque"],
    ['code' => 'BHD', 'name' => "Dinar bahreïni"],
    ['code' => 'IQD', 'name' => "Dinar irakien"],
    ['code' => 'JOD', 'name' => "Dinar jordanien"],
    ['code' => 'KWD', 'name' => "Dinar koweïtien"],
    ['code' => 'LYD', 'name' => "Dinar libyen"],
    ['code' => 'MKD', 'name' => "Dinar macédonien"],
    ['code' => 'RSD', 'name' => "Dinar serbe"],
    ['code' => 'AED', 'name' => "Dirham des Émirats arabes unis"],
    ['code' => 'USD', 'name' => "Dollar américain"],
    ['code' => 'AUD', 'name' => "Dollar australien"],
    ['code' => 'BZD', 'name' => "Dollar bélizien"],
    ['code' => 'BND', 'name' => "Dollar brunéien"],
    ['code' => 'HKD', 'name' => "Dollar de Hong Kong"],
    ['code' => 'SGD', 'name' => "Dollar de Singapour"],
    ['code' => 'TWD', 'name' => "Dollar de Taïwan"],
    ['code' => 'TTD', 'name' => "Dollar de Trinité-et-Tobago"],
    ['code' => 'JMD', 'name' => "Dollar jamaïcain"],
    ['code' => 'NAD', 'name' => "Dollar namibien"],
    ['code' => 'NZD', 'name' => "Dollar néo-zélandais"],
    ['code' => 'ZWL', 'name' => "Dollar zimbabwéen"],
    ['code' => 'VND', 'name' => "Dong vietnamien"],
    ['code' => 'AMD', 'name' => "Dram arménien"],
    ['code' => 'CVE', 'name' => "Escudo cap-verdien"],
    ['code' => 'HUF', 'name' => "Forint hongrois"],
    ['code' => 'PYG', 'name' => "Guarani paraguayen"],
    ['code' => 'UAH', 'name' => "Hryvnia ukrainienne"],
    ['code' => 'ISK', 'name' => "Króna islandaise"],
    ['code' => 'ZMW', 'name' => "Kwacha zambien"],
    ['code' => 'MMK', 'name' => "Kyat birman"],
    ['code' => 'GEL', 'name' => "Lari géorgien"],
    ['code' => 'ALL', 'name' => "Lek albanais"],
    ['code' => 'HNL', 'name' => "Lempira hondurien"],
    ['code' => 'MDL', 'name' => "Leu moldave"],
    ['code' => 'RON', 'name' => "Leu roumain"],
    ['code' => 'BGN', 'name' => "Lev bulgare"],
    ['code' => 'EGP', 'name' => "Livre égyptienne"],
    ['code' => 'LBP', 'name' => "Livre libanaise"],
    ['code' => 'SDG', 'name' => "Livre soudanaise"],
    ['code' => 'GBP', 'name' => "Livre sterling"],
    ['code' => 'SYP', 'name' => "Livre syrienne"],
    ['code' => 'TRY', 'name' => "Livre turque"],
    ['code' => 'AZN', 'name' => "Manat azerbaïdjanais"],
    ['code' => 'BAM', 'name' => "Mark convertible bosniaque"],
    ['code' => 'MZN', 'name' => "Metical mozambicain"],
    ['code' => 'NGN', 'name' => "Naira nigérian"],
    ['code' => 'ERN', 'name' => "Nakfa érythréenne"],
    ['code' => 'PEN', 'name' => "Nuevo sol péruvien"],
    ['code' => 'TOP', 'name' => "Paʻanga tongien"],
    ['code' => 'MOP', 'name' => "Pataca macanaise"],
    ['code' => 'ARS', 'name' => "Peso argentin"],
    ['code' => 'CLP', 'name' => "Peso chilien"],
    ['code' => 'COP', 'name' => "Peso colombien"],
    ['code' => 'DOP', 'name' => "Peso dominicain"],
    ['code' => 'MXN', 'name' => "Peso mexicain"],
    ['code' => 'PHP', 'name' => "Peso philippin"],
    ['code' => 'UYU', 'name' => "Peso uruguayen"],
    ['code' => 'BWP', 'name' => "Pula botswanais"],
    ['code' => 'GTQ', 'name' => "Quetzal guatémaltèque"],
    ['code' => 'ZAR', 'name' => "Rand sud-africain"],
    ['code' => 'BRL', 'name' => "Real brésilien"],
    ['code' => 'IRR', 'name' => "Rial iranien"],
    ['code' => 'OMR', 'name' => "Rial omanais"],
    ['code' => 'QAR', 'name' => "Rial qatari"],
    ['code' => 'YER', 'name' => "Rial yéménite"],
    ['code' => 'KHR', 'name' => "Riel cambodgien"],
    ['code' => 'MYR', 'name' => "Ringgit malaisien"],
    ['code' => 'SAR', 'name' => "Riyal saoudien"],
    ['code' => 'BYN', 'name' => "Rouble biélorusse"],
    ['code' => 'RUB', 'name' => "Rouble russe"],
    ['code' => 'INR', 'name' => "Roupie indienne"],
    ['code' => 'MUR', 'name' => "Roupie mauricienne"],
    ['code' => 'NPR', 'name' => "Roupie népalaise"],
    ['code' => 'PKR', 'name' => "Roupie pakistanaise"],
    ['code' => 'LKR', 'name' => "Roupie srilankaise"],
    ['code' => 'IDR', 'name' => "Rupiah indonésienne"],
    ['code' => 'ILS', 'name' => "Shekel israélien"],
    ['code' => 'KES', 'name' => "Shilling kényan"],
    ['code' => 'UGX', 'name' => "Shilling ougandais"],
    ['code' => 'SOS', 'name' => "Shilling somalien"],
    ['code' => 'TZS', 'name' => "Shilling tanzanien"],
    ['code' => 'UZS', 'name' => "Som ouzbek"],
    ['code' => 'BDT', 'name' => "Taka bangladais"],
    ['code' => 'KZT', 'name' => "Tenge kazakh"],
    ['code' => 'KRW', 'name' => "Won sud-coréen"],
    ['code' => 'JPY', 'name' => "Yen japonais"],
    ['code' => 'CNY', 'name' => "Yuan chinois"],
    ['code' => 'PLN', 'name' => "Zloty polonais"],
];
