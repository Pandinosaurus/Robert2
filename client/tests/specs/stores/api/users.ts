import requester from '@/globals/requester';
import apiUsers from '@/stores/api/users';
import { withPaginationEnvelope } from '@fixtures/@utils';
import data from '@fixtures/users';

describe('Users Api', () => {
    describe('all()', () => {
        it('parse the returned data correctly', async () => {
            const paginatedData = withPaginationEnvelope(data.default());
            jest.spyOn(requester, 'get').mockResolvedValue({ data: paginatedData });
            expect(await apiUsers.all()).toMatchSnapshot();
        });
    });

    describe('one()', () => {
        it('parse the returned data correctly', async () => {
            // - Avec lui-même.
            jest.spyOn(requester, 'get').mockResolvedValue({ data: data.details(1) });
            expect(await apiUsers.one('self')).toMatchSnapshot();

            // - Avec les autres utilisateurs.
            await Promise.all(
                data.details().map(async (datum: any) => {
                    jest.spyOn(requester, 'get').mockResolvedValue({ data: datum });
                    expect(await apiUsers.one(datum.id)).toMatchSnapshot();
                }),
            );
        });
    });

    describe('create()', () => {
        it.each(data.details())('parse the returned data correctly (with #$id)', async (datum: any) => {
            jest.spyOn(requester, 'post').mockResolvedValue({ data: datum });
            expect(await apiUsers.create({} as any)).toMatchSnapshot();
        });
    });

    describe('update()', () => {
        it('parse the returned data correctly', async () => {
            jest.spyOn(requester, 'put').mockResolvedValue({ data: data.details(1) });
            expect(await apiUsers.update('self', {} as any)).toMatchSnapshot();

            // - Avec les autres utilisateurs.
            await Promise.all(
                data.details().map(async (datum: any) => {
                    if (datum.id === 1) {
                        return;
                    }

                    jest.spyOn(requester, 'put').mockResolvedValue({ data: datum });
                    expect(await apiUsers.update(datum.id, {} as any)).toMatchSnapshot();
                }),
            );
        });
    });

    describe('getSettings()', () => {
        it('parse the returned data correctly', async () => {
            // - Avec lui-même.
            jest.spyOn(requester, 'get').mockResolvedValue({ data: data.settings(1) });
            expect(await apiUsers.getSettings('self')).toMatchSnapshot();

            // - Avec les autres utilisateurs.
            await Promise.all(
                data.settings().map(async (datum: any) => {
                    jest.spyOn(requester, 'get').mockResolvedValue({ data: datum });
                    expect(await apiUsers.getSettings(datum.id)).toMatchSnapshot();
                }),
            );
        });
    });

    describe('updateSettings()', () => {
        it('parse the returned data correctly', async () => {
            jest.spyOn(requester, 'put').mockResolvedValue({ data: data.settings(1) });
            expect(await apiUsers.updateSettings('self', {} as any)).toMatchSnapshot();

            // - Avec les autres utilisateurs.
            await Promise.all(
                data.settings().map(async (datum: any) => {
                    if (datum.id === 1) {
                        return;
                    }

                    jest.spyOn(requester, 'put').mockResolvedValue({ data: datum });
                    expect(await apiUsers.updateSettings(datum.id, {} as any)).toMatchSnapshot();
                }),
            );
        });
    });

    describe('restore()', () => {
        it.each(data.details())('parse the returned data correctly (with #$id)', async (datum: any) => {
            jest.spyOn(requester, 'put').mockResolvedValue({ data: datum });
            expect(await apiUsers.restore(datum.id)).toMatchSnapshot();
        });
    });
});
