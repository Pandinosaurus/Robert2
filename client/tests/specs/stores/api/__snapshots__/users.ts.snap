// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Api all() parse the returned data correctly 1`] = `
{
  "data": [
    {
      "email": "tester@robertmanager.net",
      "first_name": "Jean",
      "full_name": "Jean Fountain",
      "group": "administration",
      "id": 1,
      "last_name": "Fountain",
      "phone": null,
      "pseudo": "test1",
    },
    {
      "email": "tester2@robertmanager.net",
      "first_name": "Roger",
      "full_name": "Roger Rabbit",
      "group": "management",
      "id": 2,
      "last_name": "Rabbit",
      "phone": null,
      "pseudo": "test2",
    },
    {
      "email": "nobody@robertmanager.net",
      "first_name": "?",
      "full_name": "?",
      "group": "management",
      "id": 3,
      "last_name": "?",
      "phone": null,
      "pseudo": "nobody",
    },
    {
      "email": "visitor@robertmanager.net",
      "first_name": "Henry",
      "full_name": "Henry Berluc",
      "group": "readonly-planning-general",
      "id": 4,
      "last_name": "Berluc",
      "phone": "+33724000000",
      "pseudo": "TheVisitor",
    },
    {
      "email": "external@robertmanager.net",
      "first_name": "Caroline",
      "full_name": "Caroline Farol",
      "group": "readonly-planning-general",
      "id": 5,
      "last_name": "Farol",
      "phone": "+33786325500",
      "pseudo": "caroline",
    },
  ],
  "pagination": {
    "currentPage": 1,
    "perPage": 5,
    "total": {
      "items": 5,
      "pages": 1,
    },
  },
}
`;

exports[`Users Api create() parse the returned data correctly (with #1) 1`] = `
{
  "country": {
    "code": "FR",
    "id": 1,
    "name": "France",
  },
  "country_id": 1,
  "email": "tester@robertmanager.net",
  "first_name": "Jean",
  "full_address": "1, somewhere av.
1234 Megacity",
  "full_name": "Jean Fountain",
  "group": "administration",
  "id": 1,
  "last_name": "Fountain",
  "locality": "Megacity",
  "phone": null,
  "postal_code": "1234",
  "pseudo": "test1",
  "street": "1, somewhere av.",
}
`;

exports[`Users Api create() parse the returned data correctly (with #2) 1`] = `
{
  "country": null,
  "country_id": null,
  "email": "tester2@robertmanager.net",
  "first_name": "Roger",
  "full_address": null,
  "full_name": "Roger Rabbit",
  "group": "management",
  "id": 2,
  "last_name": "Rabbit",
  "locality": null,
  "phone": null,
  "postal_code": null,
  "pseudo": "test2",
  "street": null,
}
`;

exports[`Users Api create() parse the returned data correctly (with #3) 1`] = `
{
  "country": null,
  "country_id": null,
  "email": "nobody@robertmanager.net",
  "first_name": "?",
  "full_address": "156 bis, avenue des tests poussés
88080 Wazzaville",
  "full_name": "?",
  "group": "management",
  "id": 3,
  "last_name": "?",
  "locality": "Wazzaville",
  "phone": null,
  "postal_code": "88080",
  "pseudo": "nobody",
  "street": "156 bis, avenue des tests poussés",
}
`;

exports[`Users Api create() parse the returned data correctly (with #4) 1`] = `
{
  "country": {
    "code": "CH",
    "id": 2,
    "name": "Suisse",
  },
  "country_id": 2,
  "email": "visitor@robertmanager.net",
  "first_name": "Henry",
  "full_address": null,
  "full_name": "Henry Berluc",
  "group": "readonly-planning-general",
  "id": 4,
  "last_name": "Berluc",
  "locality": null,
  "phone": "+33724000000",
  "postal_code": null,
  "pseudo": "TheVisitor",
  "street": null,
}
`;

exports[`Users Api create() parse the returned data correctly (with #5) 1`] = `
{
  "country": null,
  "country_id": null,
  "email": "external@robertmanager.net",
  "first_name": "Caroline",
  "full_address": null,
  "full_name": "Caroline Farol",
  "group": "readonly-planning-general",
  "id": 5,
  "last_name": "Farol",
  "locality": null,
  "phone": "+33786325500",
  "postal_code": null,
  "pseudo": "caroline",
  "street": null,
}
`;

exports[`Users Api getSettings() parse the returned data correctly 1`] = `
{
  "default_bookings_view": "calendar",
  "language": "en",
}
`;

exports[`Users Api getSettings() parse the returned data correctly 2`] = `
{
  "default_bookings_view": "calendar",
  "language": "en",
}
`;

exports[`Users Api getSettings() parse the returned data correctly 3`] = `
{
  "default_bookings_view": "calendar",
  "language": "fr",
}
`;

exports[`Users Api getSettings() parse the returned data correctly 4`] = `
{
  "default_bookings_view": "listing",
  "language": "fr",
}
`;

exports[`Users Api getSettings() parse the returned data correctly 5`] = `
{
  "default_bookings_view": "calendar",
  "language": "fr",
}
`;

exports[`Users Api getSettings() parse the returned data correctly 6`] = `
{
  "default_bookings_view": "calendar",
  "language": "en",
}
`;

exports[`Users Api one() parse the returned data correctly 1`] = `
{
  "country": {
    "code": "FR",
    "id": 1,
    "name": "France",
  },
  "country_id": 1,
  "email": "tester@robertmanager.net",
  "first_name": "Jean",
  "full_address": "1, somewhere av.
1234 Megacity",
  "full_name": "Jean Fountain",
  "group": "administration",
  "id": 1,
  "last_name": "Fountain",
  "locality": "Megacity",
  "phone": null,
  "postal_code": "1234",
  "pseudo": "test1",
  "street": "1, somewhere av.",
}
`;

exports[`Users Api one() parse the returned data correctly 2`] = `
{
  "country": {
    "code": "FR",
    "id": 1,
    "name": "France",
  },
  "country_id": 1,
  "email": "tester@robertmanager.net",
  "first_name": "Jean",
  "full_address": "1, somewhere av.
1234 Megacity",
  "full_name": "Jean Fountain",
  "group": "administration",
  "id": 1,
  "last_name": "Fountain",
  "locality": "Megacity",
  "phone": null,
  "postal_code": "1234",
  "pseudo": "test1",
  "street": "1, somewhere av.",
}
`;

exports[`Users Api one() parse the returned data correctly 3`] = `
{
  "country": null,
  "country_id": null,
  "email": "tester2@robertmanager.net",
  "first_name": "Roger",
  "full_address": null,
  "full_name": "Roger Rabbit",
  "group": "management",
  "id": 2,
  "last_name": "Rabbit",
  "locality": null,
  "phone": null,
  "postal_code": null,
  "pseudo": "test2",
  "street": null,
}
`;

exports[`Users Api one() parse the returned data correctly 4`] = `
{
  "country": null,
  "country_id": null,
  "email": "nobody@robertmanager.net",
  "first_name": "?",
  "full_address": "156 bis, avenue des tests poussés
88080 Wazzaville",
  "full_name": "?",
  "group": "management",
  "id": 3,
  "last_name": "?",
  "locality": "Wazzaville",
  "phone": null,
  "postal_code": "88080",
  "pseudo": "nobody",
  "street": "156 bis, avenue des tests poussés",
}
`;

exports[`Users Api one() parse the returned data correctly 5`] = `
{
  "country": {
    "code": "CH",
    "id": 2,
    "name": "Suisse",
  },
  "country_id": 2,
  "email": "visitor@robertmanager.net",
  "first_name": "Henry",
  "full_address": null,
  "full_name": "Henry Berluc",
  "group": "readonly-planning-general",
  "id": 4,
  "last_name": "Berluc",
  "locality": null,
  "phone": "+33724000000",
  "postal_code": null,
  "pseudo": "TheVisitor",
  "street": null,
}
`;

exports[`Users Api one() parse the returned data correctly 6`] = `
{
  "country": null,
  "country_id": null,
  "email": "external@robertmanager.net",
  "first_name": "Caroline",
  "full_address": null,
  "full_name": "Caroline Farol",
  "group": "readonly-planning-general",
  "id": 5,
  "last_name": "Farol",
  "locality": null,
  "phone": "+33786325500",
  "postal_code": null,
  "pseudo": "caroline",
  "street": null,
}
`;

exports[`Users Api restore() parse the returned data correctly (with #1) 1`] = `
{
  "country": {
    "code": "FR",
    "id": 1,
    "name": "France",
  },
  "country_id": 1,
  "email": "tester@robertmanager.net",
  "first_name": "Jean",
  "full_address": "1, somewhere av.
1234 Megacity",
  "full_name": "Jean Fountain",
  "group": "administration",
  "id": 1,
  "last_name": "Fountain",
  "locality": "Megacity",
  "phone": null,
  "postal_code": "1234",
  "pseudo": "test1",
  "street": "1, somewhere av.",
}
`;

exports[`Users Api restore() parse the returned data correctly (with #2) 1`] = `
{
  "country": null,
  "country_id": null,
  "email": "tester2@robertmanager.net",
  "first_name": "Roger",
  "full_address": null,
  "full_name": "Roger Rabbit",
  "group": "management",
  "id": 2,
  "last_name": "Rabbit",
  "locality": null,
  "phone": null,
  "postal_code": null,
  "pseudo": "test2",
  "street": null,
}
`;

exports[`Users Api restore() parse the returned data correctly (with #3) 1`] = `
{
  "country": null,
  "country_id": null,
  "email": "nobody@robertmanager.net",
  "first_name": "?",
  "full_address": "156 bis, avenue des tests poussés
88080 Wazzaville",
  "full_name": "?",
  "group": "management",
  "id": 3,
  "last_name": "?",
  "locality": "Wazzaville",
  "phone": null,
  "postal_code": "88080",
  "pseudo": "nobody",
  "street": "156 bis, avenue des tests poussés",
}
`;

exports[`Users Api restore() parse the returned data correctly (with #4) 1`] = `
{
  "country": {
    "code": "CH",
    "id": 2,
    "name": "Suisse",
  },
  "country_id": 2,
  "email": "visitor@robertmanager.net",
  "first_name": "Henry",
  "full_address": null,
  "full_name": "Henry Berluc",
  "group": "readonly-planning-general",
  "id": 4,
  "last_name": "Berluc",
  "locality": null,
  "phone": "+33724000000",
  "postal_code": null,
  "pseudo": "TheVisitor",
  "street": null,
}
`;

exports[`Users Api restore() parse the returned data correctly (with #5) 1`] = `
{
  "country": null,
  "country_id": null,
  "email": "external@robertmanager.net",
  "first_name": "Caroline",
  "full_address": null,
  "full_name": "Caroline Farol",
  "group": "readonly-planning-general",
  "id": 5,
  "last_name": "Farol",
  "locality": null,
  "phone": "+33786325500",
  "postal_code": null,
  "pseudo": "caroline",
  "street": null,
}
`;

exports[`Users Api update() parse the returned data correctly 1`] = `
{
  "country": {
    "code": "FR",
    "id": 1,
    "name": "France",
  },
  "country_id": 1,
  "email": "tester@robertmanager.net",
  "first_name": "Jean",
  "full_address": "1, somewhere av.
1234 Megacity",
  "full_name": "Jean Fountain",
  "group": "administration",
  "id": 1,
  "last_name": "Fountain",
  "locality": "Megacity",
  "phone": null,
  "postal_code": "1234",
  "pseudo": "test1",
  "street": "1, somewhere av.",
}
`;

exports[`Users Api update() parse the returned data correctly 2`] = `
{
  "country": null,
  "country_id": null,
  "email": "tester2@robertmanager.net",
  "first_name": "Roger",
  "full_address": null,
  "full_name": "Roger Rabbit",
  "group": "management",
  "id": 2,
  "last_name": "Rabbit",
  "locality": null,
  "phone": null,
  "postal_code": null,
  "pseudo": "test2",
  "street": null,
}
`;

exports[`Users Api update() parse the returned data correctly 3`] = `
{
  "country": null,
  "country_id": null,
  "email": "nobody@robertmanager.net",
  "first_name": "?",
  "full_address": "156 bis, avenue des tests poussés
88080 Wazzaville",
  "full_name": "?",
  "group": "management",
  "id": 3,
  "last_name": "?",
  "locality": "Wazzaville",
  "phone": null,
  "postal_code": "88080",
  "pseudo": "nobody",
  "street": "156 bis, avenue des tests poussés",
}
`;

exports[`Users Api update() parse the returned data correctly 4`] = `
{
  "country": {
    "code": "CH",
    "id": 2,
    "name": "Suisse",
  },
  "country_id": 2,
  "email": "visitor@robertmanager.net",
  "first_name": "Henry",
  "full_address": null,
  "full_name": "Henry Berluc",
  "group": "readonly-planning-general",
  "id": 4,
  "last_name": "Berluc",
  "locality": null,
  "phone": "+33724000000",
  "postal_code": null,
  "pseudo": "TheVisitor",
  "street": null,
}
`;

exports[`Users Api update() parse the returned data correctly 5`] = `
{
  "country": null,
  "country_id": null,
  "email": "external@robertmanager.net",
  "first_name": "Caroline",
  "full_address": null,
  "full_name": "Caroline Farol",
  "group": "readonly-planning-general",
  "id": 5,
  "last_name": "Farol",
  "locality": null,
  "phone": "+33786325500",
  "postal_code": null,
  "pseudo": "caroline",
  "street": null,
}
`;

exports[`Users Api updateSettings() parse the returned data correctly 1`] = `
{
  "default_bookings_view": "calendar",
  "language": "en",
}
`;

exports[`Users Api updateSettings() parse the returned data correctly 2`] = `
{
  "default_bookings_view": "calendar",
  "language": "en",
}
`;

exports[`Users Api updateSettings() parse the returned data correctly 3`] = `
{
  "default_bookings_view": "calendar",
  "language": "fr",
}
`;

exports[`Users Api updateSettings() parse the returned data correctly 4`] = `
{
  "default_bookings_view": "listing",
  "language": "fr",
}
`;

exports[`Users Api updateSettings() parse the returned data correctly 5`] = `
{
  "default_bookings_view": "calendar",
  "language": "fr",
}
`;

exports[`Users Api updateSettings() parse the returned data correctly 6`] = `
{
  "default_bookings_view": "calendar",
  "language": "en",
}
`;
